src/main/java/core/mate/academy/service/ExcavatorProducerImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducerImpl<T extends Machine> extends TrackProducerImpl {
```

Comment:
```suggestion
public class ExcavatorProducer implements MachineProducer< Excavator> {
```


why ExcavatorProducerImpl implements TrackProducerImpl?